habitat/datasets/rearrange/samplers.py

```diff
@@ -577,9 +615,18 @@ def sample(
                 # NOTE: only query and set pose once per instance for efficiency
                 pose = ao_instance.joint_positions
                 for link_ix, joint_range in link_ranges.items():
-                    joint_state = random.uniform(
-                        joint_range[0], joint_range[1]
-                    )
+                    if (
+                        ao_instance.handle == receptacle.parent_object_handle
+                    ) and (
+                        (link_ix == receptacle.parent_link)
+                        or ("frige" in receptacle.parent_object_handle)
+                        or ("fridge" in receptacle.parent_object_handle)
+                    ):
+                        joint_state = random.uniform(
+                            joint_range[0], joint_range[1]
+                        )
+                    else:
+                        joint_state = 0.0
```

Comment:
Here again we are restricting the general behavior of the sampler with a hardcoded override to both require and only consider a single receptacle. This should be redesigned with a configuration option so it is less contrived.